### ISSUE TITLE
Issue with re-assigning exported local variable after declaration.

### ISSUE THREAD

**Issue by rwjblue** [esperantojs/esperanto#130]
Input:

```javascript
var foo = function() { }

if (false) {
  foo = function () { };  
}

export {
 foo 
}
```

Output:

```javascript
define(['exports'], function (exports) {

  'use strict';

  var foo = function() { }

  if (false) {
    exports.foo = foo = function () { };  
  }

});
```

---

This causes some issues in Ember, and we have locked Esperanto down to 0.6.17 temporarily (in https://github.com/emberjs/ember.js/pull/10751).

**Comment by Rich-Harris** [comment]
Ah, yep, I think I see the problem. Thinking out loud: it's trying to avoid doing more `exports.foo = ...` than is necessary - [this line](https://github.com/esperantojs/esperanto/blob/a6a9ad9dbfc7e1e29f8a57ca0f0aa3f9e9107e78/src/utils/ast/rewriteExportAssignments.js#L43) is being run, because `isTopLevelNode` is true but shouldn't be. The problematic line is [here](https://github.com/esperantojs/esperanto/blob/57831f9c3b6f5104cc2c856c2a8c6cfbec17b8b3/src/utils/ast/traverse.js#L40) - it's determining whether it's a top level node by seeing if it's in the top level *scope*, which obviously isn't the case since it's in a block. Should be an easy fix - thanks for raising it

**Comment by eventualbuddha** [comment]
I believe the expected output should include two `exports.foo` assignments, one inside the `if`, the other outside it.

**Comment by rwjblue** [comment]
@eventualbuddha - OK, I definitely defer that to y'all. My main issue is that there is no `export.foo` in the current output (because it is inside the `if`).  I'll just remove the expected output....

**Comment by Rich-Harris** [comment]
Have opened #131, which fixes this but changes the approach slightly. Previously, esperanto would keep track of any assignments to exported bindings, so that it didn't need the final `exports.foo = foo`. A simpler approach in my view is to *always* have `exports.foo = foo` at the end, and not bother with changing the `foo = ...` into `exports.foo = foo = ...` inside the `if` block.

The one exception to this is when `foo` is assigned to inside a function body, since that could happen late. So for example [this](https://github.com/esperantojs/esperanto/blob/gh-130/test/samples/conditionalExport/source.js) becomes [this](https://github.com/esperantojs/esperanto/blob/gh-130/test/strictMode/output/amd/conditionalExport.js).

Does that sound right to everyone?

**Comment by Rich-Harris** [comment]
Released in 0.6.23
